init.d/urandom.in

```diff
@@ -24,35 +21,33 @@ depend()
 
 save_seed()
 {
+	local psz=1
+
+	if [ -e /proc/sys/kernel/random/poolsize ]; then
+		: $(( psz = $(cat /proc/sys/kernel/random/poolsize) / 4096 ))
+	fi
+
 	(	# sub shell to prevent umask pollution
 		umask 077
-		dd if=/dev/urandom of="$urandom_seed" count=1 2>/dev/null
+		dd if=/dev/urandom of="$urandom_seed" count=${psz} 2>/dev/null
 	)
 }
 
 start()
 {
-	if [ "$RC_UNAME" = Linux ]; then
-		seedrng
-	else
-		[ -c /dev/urandom ] || return
-		if [ -f "$urandom_seed" ]; then
-			ebegin "Initializing random number generator"
-			cat "$urandom_seed" > /dev/urandom
-			eend $? "Error initializing random number generator"
-		fi
-		rm -f "$urandom_seed" && save_seed
+	[ -c /dev/urandom ] || return
+	if [ -f "$urandom_seed" ]; then
+		ebegin "Initializing random number generator"
+		cat "$urandom_seed" > /dev/urandom
+		eend $? "Error initializing random number generator"
 	fi
+	rm -f "$urandom_seed" && save_seed
```

Comment:
Delete after reading but before seeding. And issue an fsync.

Reply:
I just did a quick check via google and was unable to find an "fsync" command that can be run from the shell on *BSD. For now, I'm just looking to take urandom back to what it was before it was wired to seedrng, but we can improve it in a separate pr.

Reply:
FreeBSD comes with `sync(8)`, but that doesn't take a file argument, so yea, I guess we can table this until later.